commitizen/git.py

```diff
@@ -82,13 +82,16 @@ def from_line(cls, line: str, inner_delimiter: str) -> "GitTag":
         return cls(name=name, rev=obj, date=date)
 
 
-def tag(tag: str, annotated: bool = False, signed: bool = False) -> cmd.Command:
+def tag(
+    tag: str, annotated: bool = False, signed: bool = False, msg: Optional[str] = None
+) -> cmd.Command:
     _opt = ""
     if annotated:
         _opt = f"-a {tag} -m"
     if signed:
         _opt = f"-s {tag} -m"
-    c = cmd.run(f"git tag {_opt} {tag}")
+
+    c = cmd.run(f'git tag {_opt} "{tag if _opt == "" or msg is None else msg}"')
```

Comment:
Does that mean if we provide `msg` then we won't have `tag?`

Reply:
No, the parameter `tag` is mandatory. So we will have a tag nonetheless.
If we provide a `msg`, it will all depend on whether we defined the parameters `annotated` or `signed`. Because if `msg` is defined, the `or msg is None` part will be False and it will depend on `_opt`, which depends on `annotated` or `signed`.

Assuming we HAVE defined `msg` and `tag=MY_TAG`:
1. If we define `annotated = False` or `signed = False`, `_opt` will remain an empty string.
   - This means the condition `_opt == ""` will be True, so: `git tag "MY_TAG"`
2. If we define either `annotated = True` or `signed = True`, `_opt` won't be an empty string anymore.
   - This means the condition `_opt == ""` will be False. Since `msg` is also defined, `msg is None` will be False as well. So, the resulting command will be, for example: `git tag -a MY_TAG -m {msg}`